datasetinsights/data/download.py

```diff
@@ -64,3 +65,47 @@ def download_file(source_uri: str, dest_path: str, use_cache: bool = True):
                 f.write(response.content)
 
     return dest_path
+
+
+def validate_checksum(filepath, expected_checksum, algorithm="CRC32"):
+    """ Validate checksum of the downloaded file.
+
+    Args:
+        filepath (str): the doaloaded file path
+        expected_checksum (int): expected checksum of the file
+        algorithm (str): checksum algorithm. Defaults to CRC32
+
+    Raises:
+        ChecksumError if the file checksum does not match.
+    """
+    computed = compute_checksum(filepath, algorithm)
+    if computed != expected_checksum:
+        raise ChecksumError
+
+
+def compute_checksum(filepath, algorithm="CRC32"):
+    """ Compute the checksum of a file.
+
+    Args:
+        filepath (str): the doaloaded file path
+        algorithm (str): checksum algorithm. Defaults to CRC32
+
+    Returns:
+        int: the checksum value
+    """
+    chs = -1
```

Comment:
Why do we need a default checksum value `-1`? If the checksum computation failed, doesn't it simply raise some error?

Reply:
Yes, it does. Will remove the initialization in case there is a checksum of a file equals -1.